src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -701,6 +709,32 @@ private FunctionCompletedMessage CreateStartedMessage(IFunctionInstance instance
             return message;
         }
 
+#nullable enable
+        private IActivityAbstraction? TryCreateFunctionActivity(IFunctionInstanceEx functionInstanceEx)
+        {
+            // Return null if no activity source is provided
+            if (_activitySource == null || _activityContextProvider == null)
+            {
+                return null;
+            }
+
+            // If no current activity exists, create one for the entire function run.
+            // HTTP, Service Bus, Event Hub, and other instrumented triggers will have their own activities.
+            // BeginFunctionScope creates a function activity when AppInsights SDK is enabled.
+            // In OTel mode, Activity.Current will be null unless the trigger is instrumented.
+            if (_activityContextProvider.HasCurrentActivity)
+            {
+                return null;
+            }
+
+            // Start new activity for the function execution
+            return _activitySource.StartActivity(
```

Comment:
I wonder if the `IActivitySourceAbstraction.StartActivity` should take in `IFunctionInstanceEx` and let the implementation decide what properties to use.